nav2_world_model/src/main.cpp

```diff
@@ -19,7 +19,10 @@
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  rclcpp::spin(std::make_shared<nav2_world_model::WorldModel>());
+  rclcpp::executors::SingleThreadedExecutor exec;
```

Comment:
I see that the implementation of WorldModel needs to get the executor because it creates another ROS node (the costmap_2d_ros node) and adds it to the executor. Too bad we can't get the executor another way so that the "outside" code (this code here in main) doesn't need to treat this node differently just because the implementation uses another ROS node. In other words, it would be better to use typical pattern in main:

`
rclcpp::init(argc, argv);
rclcpp::spin(std::make_shared<nav2_world_model::WorldModel>());
rclcpp::shutdown();
`

And have the WorldModel node get its executor another way. However, I don't see a way for a node to get its own executor so that any sub-nodes could be added to it.

Reply:
I think Carl suggested at some point that we could perhaps add a function to return the node shared pointer. Then, we could spin that node from main with the executor. I'm not sure if that is considered a better design, what do you think?